rotkehlchen/chain/evm/decoding/curve/curve_cache.py

```diff
@@ -388,8 +408,21 @@ def _query_curve_data_from_chain(
         deployed_block=0,  # deployment_block is not used and the contract is dynamic
     )
     pool_count = metaregistry.call(node_inquirer=evm_inquirer, method_name='pool_count')
-    new_pools = []
+    new_pools: list[CurvePoolData] = []
+    last_notified_ts = 0
     for pool_index in range(pool_count):
+        if (current_time := ts_now()) - last_notified_ts > 5:
+            msg_aggregator.add_message(
+                message_type=WSMessageType.PROTOCOL_CACHE_UPDATES,
+                data={
+                    'protocol': CPT_CURVE,
+                    'chain': evm_inquirer.chain_id.to_name(),
+                    'processed': 0,
```

Comment:
notice how you are sending 0 here always

Reply:
in some places the `processed` is 0 because 0 pools are processed at that point. so, it happens in two iterations.
first we iterate to find how many new pools are there, this is where we populate the `total` value and will increment it at every message sent.
then in the next iteration `total` number is constant, and `processed` is incremented.